src/test/java/seedu/address/storage/JsonAdaptedEventTest.java

```diff
@@ -0,0 +1,172 @@
+        JsonAdaptedEvent event = new JsonAdaptedEvent(
+                VALID_EVENT_DESCRIPTION, null, VALID_TIME, VALID_DURATION, VALID_RECURFREQUENCY);
+        String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, LocalDate.class.getSimpleName());
+        assertThrows(IllegalValueException.class, expectedMessage, event::toModelType);
+    }
+
+    @Test
+    public void toModelType_invalidTime_throwsIllegalValueException() {
+        JsonAdaptedEvent event = new JsonAdaptedEvent(
+                VALID_EVENT_DESCRIPTION, VALID_DATE, INVALID_TIME, VALID_DURATION, VALID_RECURFREQUENCY);
+        String expectedMessage = Event.TIME_MESSAGE_CONSTRAINTS;
+        assertThrows(IllegalValueException.class, expectedMessage, event::toModelType);
+    }
+
+    @Test
+    public void toModelType_nullTime_throwsIllegalValueException() {
+        JsonAdaptedEvent event = new JsonAdaptedEvent(
+                VALID_EVENT_DESCRIPTION, VALID_DATE, null, VALID_DURATION, VALID_RECURFREQUENCY);
+        String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, LocalTime.class.getSimpleName());
+        assertThrows(IllegalValueException.class, expectedMessage, event::toModelType);
+    }
+
+    @Test
+    public void toModelType_invalidDuration_throwsIllegalValueException() {
+        JsonAdaptedEvent event = new JsonAdaptedEvent(
+                VALID_EVENT_DESCRIPTION, VALID_DATE, VALID_TIME, INVALID_DURATION, VALID_RECURFREQUENCY);
+        String expectedMessage = Event.DURATION_MESSAGE_CONSTRAINTS;
+        assertThrows(IllegalValueException.class, expectedMessage, event::toModelType);
+    }
+
+    @Test
+    public void toModelType_nullDuration_throwsIllegalValueException() {
+        JsonAdaptedEvent event = new JsonAdaptedEvent(
+                VALID_EVENT_DESCRIPTION, VALID_DATE, VALID_TIME, null, VALID_RECURFREQUENCY);
+        String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Duration.class.getSimpleName());
+        assertThrows(IllegalValueException.class, expectedMessage, event::toModelType);
+    }
+
+    @Test
+    public void toModelType_invalidRecurFrequency_throwsIllegalValueException() {
+        JsonAdaptedEvent event = new JsonAdaptedEvent(
+                VALID_EVENT_DESCRIPTION, VALID_DATE, VALID_TIME, VALID_DURATION, INVALID_RECURFREQUENCY);
+        String expectedMessage = RecurFrequency.INVALID_RECUR_FREQUENCY_MESSAGE;
+        assertThrows(IllegalValueException.class, expectedMessage, event::toModelType);
+    }
+
+    @Test
+    public void toModelType_nullRecurFrequency_throwsIllegalValueException() {
+        JsonAdaptedEvent event = new JsonAdaptedEvent(
+                VALID_EVENT_DESCRIPTION, VALID_DATE, VALID_TIME, VALID_DURATION, null);
+        String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, RecurFrequency.class.getSimpleName());
+        assertThrows(IllegalValueException.class, expectedMessage, event::toModelType);
+    }
```

Comment:
I think that all this methods can combined. And the name of the test method can be called `toModelType_invalidInput_throwsIllegalValueException`. Also perhaps we can use the constructor for JsonAdaptedEvent that takes in Event just for testing? Then it would be much easier to test this with EventBuilder.

Reply:
I think the functions can stay uncombined? Different parts of the AB3 does it differently, I just thought it would be best to follow closely to the other JsonAdapted class tests. (Maybe the other JsonAdapted tests were split so that the one function doesn't become too long?) Anyway, one upside of using separate test functions is also that each test is also easier to differentiate what each function is testing, and in the future if we add more different invalid Date and Time formats it won't convolute the function also.

Reply:
I suggested this because of the equivalence partition, it seems to me that they are the same just that we would have to test more the boundaries. What we can do to differentiate each function would to put some comments before each section. But if this is how the other JsonAdapted did the test, I guess it we can leave it as it is.

Reply:
Ah I see